tests/net/lib/lwm2m/content_plain_text/testcase.yaml

```diff
@@ -1,8 +1,8 @@
-common:
-  depends_on: netif
 tests:
   net.lwm2m.content_plain_text:
-    filter: TOOLCHAIN_HAS_NEWLIB == 1
-    tags:
-      - lwm2m
-      - net
```

Comment:
why is this not net related anymore? We have many platforms ignoring networking by tags, now all of thos will be building lwm2m for no good reason

Reply:
I don't think there is any reason to filter out the `net` tagged tests on platforms like `qemu_cortex_m3` that is capable of running Network tests and we already provide configuration for it in LwM2M sample.
Seems pretty useless to filter away one of our mainly used test platform.

But because that `net` filtering exist, I have to workaround it.
None of these tests actually require networking, sockets are stubbed away here. But I added filtering so that it runs only on simulated platforms so we should not waste HW resources.
Is this not sufficient enough?

Can you suggest alternative? How can I run network related test on `qemu_cortex_m3` platform?

Reply:
why do you need to run on qemu_cortex_m3? it is already covered by other emulation platforms

Reply:
One native X86 compiler (native_posix), one Cortex-M. I think it makes sense.

If not qemu_cortex_m3 what would you suggest?

Reply:
@nashif What do you mean by "now all of thos will be building lwm2m for no good reason"? I am referring to a fact that just because it can be build it doesn't mean it will be. As pointed in my previous comment, `platform_key` significantly limits the resulting builds. I thought that was the reason behind introduction of `platform_key`? (cc @teburd) Like I wrote, even with `--all` in the CLI the chosen scenario was built only on 4 boards.
I also just checked that adding platform_key benefits our on-target setup where we have ~10 different nrf boards. Without the change a single test is repeated on 3 boards (which has net capability). With the PR build is limited to a single one.

Reply:
> @nashif What do you mean by "now all of thos will be building lwm2m for no good reason"? I am referring to a fact that just because it can be build it doesn't mean it will be.

with this PR:

```
gale:zephyr(lwm2m_refactor_tests): ./scripts/twister -p qemu_xtensa -T tests/net/lib/lwm2m/ -v
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.1
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.4.0-247-g1b1ce125b39e
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 112
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue
INFO    -  1/12 qemu_xtensa               tests/net/lib/lwm2m/lwm2m_engine/net.lwm2m.lwm2m_engine PASSED (qemu 2.075s)
INFO    -  2/12 qemu_xtensa               tests/net/lib/lwm2m/lwm2m_rd_client/net.lwm2m.lwm2m_rd_client_dtls PASSED (qemu 2.273s)
INFO    -  3/12 qemu_xtensa               tests/net/lib/lwm2m/lwm2m_rd_client/net.lwm2m.lwm2m_rd_client PASSED (qemu 2.275s)
INFO    -  4/12 qemu_xtensa               tests/net/lib/lwm2m/content_json/net.lwm2m.content_json PASSED (qemu 2.118s)
INFO    -  5/12 qemu_xtensa               tests/net/lib/lwm2m/content_raw_cbor/net.lwm2m.content_raw_cbor PASSED (qemu 2.082s)
INFO    -  6/12 qemu_xtensa               tests/net/lib/lwm2m/block_transfer/net.lwm2m.block_transfer PASSED (qemu 2.106s)
INFO    -  7/12 qemu_xtensa               tests/net/lib/lwm2m/lwm2m_registry/net.lwm2m.lwm2m_registry PASSED (qemu 2.061s)
INFO    -  8/12 qemu_xtensa               tests/net/lib/lwm2m/content_oma_tlv/net.lwm2m.content_oma_tlv PASSED (qemu 2.082s)
INFO    -  9/12 qemu_xtensa               tests/net/lib/lwm2m/content_link_format/net.lwm2m.content_link_format PASSED (qemu 2.061s)
INFO    - 10/12 qemu_xtensa               tests/net/lib/lwm2m/content_plain_text/net.lwm2m.content_plain_text PASSED (qemu 2.078s)
INFO    - 11/12 qemu_xtensa               tests/net/lib/lwm2m/engine/net.lwm2m.engine        PASSED (qemu 2.068s)
INFO    - 12/12 qemu_xtensa               tests/net/lib/lwm2m/content_senml_cbor/net.lwm2m.content_senml_cbor PASSED (qemu 2.129s)

INFO    - 12 test scenarios (12 test instances) selected, 0 configurations skipped (0 by static filter, 0 at runtime).
INFO    - 12 of 12 test configurations passed (100.00%), 0 failed, 0 errored, 0 skipped with 0 warnings in 18.89 seconds
INFO    - In total 233 test cases were executed, 0 skipped on 1 out of total 575 platforms (0.17%)
INFO    - 12 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```

Without this PR:
```
gale:zephyr(main): ./scripts/twister -p qemu_xtensa -T tests/net/lib/lwm2m/ -v
Renaming output directory to /home/nashif/zephyrproject/zephyr/twister-out.2
INFO    - Using Ninja..
INFO    - Zephyr version: zephyr-v3.4.0-275-g71b522cfdd01
INFO    - Using 'zephyr' toolchain.
INFO    - Building initial testsuite list...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/testplan.json
INFO    - JOBS: 112
INFO    - Adding tasks to the queue...
INFO    - Added initial list of jobs to queue

INFO    - 12 test scenarios (12 test instances) selected, 12 configurations skipped (12 by static filter, 0 at runtime).
INFO    - 0 of 12 test configurations passed (0.00%), 0 failed, 0 errored, 12 skipped with 0 warnings in 2.08 seconds
INFO    - In total 0 test cases were executed, 233 skipped on 1 out of total 575 platforms (0.17%)
INFO    - 0 test configurations executed on platforms, 0 test configurations were only built.
INFO    - Saving reports...
INFO    - Writing JSON report /home/nashif/zephyrproject/zephyr/twister-out/twister.json
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister.xml...
INFO    - Writing xunit report /home/nashif/zephyrproject/zephyr/twister-out/twister_report.xml...
INFO    - Run completed
```


Removing the `net` tag seems to be a workaround to run tests on a platform we have not used for running network tests.

> As pointed in my previous comment, platform_key significantly limits the resulting builds. 

Where exactly did I say anything about `platform_key`?

Reply:
> If not qemu_cortex_m3 what would you suggest?

The same qemu we use for all other network based tests, i.e. qemu_x86, which also has userspace support, so it covers more than just the basics. 

> Edit: I would prefer to use same compiler `arm-zephyr-eabi` than our target HW, but without wasting HW resources.

isn't lwm2m and network in general hardware agnostic?

If you were testing some hardware features, sure, but testing networking does not really care what compiler and hardware you use AFAIK.

Reply:
Can you run the same with `native_posix` and without this PR?

I really don't see what is the point of limiting tests on platforms where they work just fine. The integration platforms are listed. According to documentation this should be the field used to limit the CI load.

Reply:
> Removing the net tag seems to be a workaround to run tests on a platform we have not used for running network tests.

Sure, but why do you block anyone from running module tests on those platforms? And what `arm-zephyr-eabi` platforms do we use for running network related tests?

Reply:
> Sure, but why do you block anyone from running module tests on those platforms?

time, money and common sense.

If you want to run those test on a specific platform and you have a good reason for that, we should change the platform not to ignore the tag, but do not remove the tag from the tests, because the lwm2m is part of networking and has always been, so do not end up building the same code aimlessly on each platform and in each PR if it can be covered by one.


> And what `arm-zephyr-eabi` platforms do we use for running network related tests?

The question is if such tests need to be built and run with each compiler, I do not think so. This should be hardware agnostic at least on the emulation side. I am sure you get to build and run those on the target hardware as well outside of project CI, so you have the coverage for ARM or any other architecture as well.

Reply:
I'm trying to suggest a sensible filtering so that this does not build on all.
And it is not sensible to assume that compiler architectures have differencies. So if Arm is our target HW, why not build tests with `arm-zephyr-eabi`. After all, that is installed on all developers on my organization.

Then bulk of testing I would prefer to use native_posix just because it is fastest and behaves nicely on GDB on local machine.

But what do you suggest? Why qemu_x86? Why install a toolchain just to run tests on architecture that matches neither the target HW or local development machine.